crates/sui-framework/packages/sui-framework/tests/random_tests.move

```diff
@@ -26,11 +33,571 @@ module sui::random_tests {
+        test_scenario::end(scenario_val);
+
+        // Set random to global_random1
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+        let random_state = test_scenario::take_shared<Random>(scenario);
+        update_randomness_state_for_testing(
+            &mut random_state,
+            0,
+            global_random1,
+            test_scenario::ctx(scenario)
+        );
+        test_scenario::next_tx(scenario, @0x0);
+        let gen1= new_generator(&random_state, test_scenario::ctx(scenario));
+        test_scenario::return_shared(random_state);
+        test_scenario::end(scenario_val);
+
+        // Set random again to global_random1
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+        let random_state = test_scenario::take_shared<Random>(scenario);
+        update_randomness_state_for_testing(
+            &mut random_state,
+            1,
+            global_random1,
+            test_scenario::ctx(scenario)
+        );
+        test_scenario::next_tx(scenario, @0x0);
+        let gen2= new_generator(&random_state, test_scenario::ctx(scenario));
+        test_scenario::return_shared(random_state);
+        test_scenario::end(scenario_val);
+
+        // Set random to global_random2
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+        let random_state = test_scenario::take_shared<Random>(scenario);
+        update_randomness_state_for_testing(
+            &mut random_state,
+            2,
+            global_random2,
+            test_scenario::ctx(scenario)
+        );
+        test_scenario::next_tx(scenario, @0x0);
+        let gen3= new_generator(&random_state, test_scenario::ctx(scenario));
+        let gen4= new_generator(&random_state, test_scenario::ctx(scenario));
+        test_scenario::return_shared(random_state);
+        test_scenario::end(scenario_val);
+
+        assert!(generator_counter(&gen1) == 0, 0);
+        assert!(*generator_buffer(&gen1) == vector::empty(), 0);
+        assert!(generator_seed(&gen1) == generator_seed(&gen2), 0);
+        assert!(generator_seed(&gen1) != generator_seed(&gen3), 0);
+        assert!(generator_seed(&gen3) != generator_seed(&gen4), 0);
+    }
+
+    #[test]
+    fun random_tests_regression() {
+        let scenario_val = test_scenario::begin(@0x0);
+        let scenario = &mut scenario_val;
+
+        random::create_for_testing(test_scenario::ctx(scenario));
+        test_scenario::next_tx(scenario, @0x0);
+
+        let random_state = test_scenario::take_shared<Random>(scenario);
+        update_randomness_state_for_testing(
+            &mut random_state,
+            0,
+            x"1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F",
+            test_scenario::ctx(scenario)
+        );
+
+        // Regression (not critical for security, but still an indication that something is wrong).
```

Comment:
how does this indicate that something is wrong? could we not change the algorithm for generating randomness in a way that causes these numbers to change but still doesn't mean anything is wrong?

Reply:
We can, it's mainly here to indicate to developers that some change, somewhere (can be even in the way we derive UIDs) impacts the randomness generator and worth noticing explicitly. In most cases it doesn't imply that something is broken, and only needs an update.